stdlib/public/Cxx/CxxSequence.swift

```diff
@@ -0,0 +1,67 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Bridged C++ iterator that allows to traverse the elements of a sequence using a for-in loop.
+///
+/// Mostly useful for conforming a type to the `CxxSequence` protocol and should not generally be used directly.
+///
+/// - SeeAlso: https://en.cppreference.com/w/cpp/named_req/InputIterator
+public protocol UnsafeCxxInputIterator: Equatable {
+  associatedtype Pointee
+
+  /// Returns the unwrapped result of C++ `operator*()`.
+  ///
+  /// Generally, Swift creates this property automatically for C++ types that define `operator*()`.
+  var pointee: Pointee { get }
+
+  /// Returns an iterator pointing to the next item in the sequence.
+  func successor() -> Self
```

Comment:
Name chosen to match `UnsafePointer.successor()`/`UnsafeMutablePointer.successor()`: I think we should try to avoid adding public methods/properties to Swift stdlib classes as much as possible.